File: src/test/java/org/hisp/dhis/rules/models/RuleVariableCurrentEventTest.java
package org.hisp.dhis.rules.models;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import java.util.ArrayList;

import static org.junit.Assert.assertEquals;

@RunWith( JUnit4.class )
public class RuleVariableCurrentEventTest
{
    @Test
    public void createShouldPropagatePropertiesCorrectly()
    {
        RuleVariableCurrentEvent ruleVariableCurrentEvent = RuleVariableCurrentEvent.create(
            "test_variable", "test_dataelement", RuleValueType.NUMERIC, true, new ArrayList<>());

        assertEquals( "test_variable" , ruleVariableCurrentEvent.name() );
        assertEquals( "test_dataelement" , ruleVariableCurrentEvent.dataElement() );
        assertEquals( RuleValueType.NUMERIC , ruleVariableCurrentEvent.dataElementType() );
    }
}

File: src/jvmTest/kotlin/org/hisp/dhis/rules/models/RuleVariableCurrentEventTest.kt
package org.hisp.dhis.rules.models

import kotlin.test.Test
import kotlin.test.assertEquals

class RuleVariableCurrentEventTest {
    @Test
    fun createShouldPropagatePropertiesCorrectly() {
        val ruleVariableCurrentEvent = RuleVariableCurrentEvent(
            "test_variable", true, ArrayList(), "test_dataelement", RuleValueType.NUMERIC
        )
        assertEquals("test_variable", ruleVariableCurrentEvent.name)
        assertEquals("test_dataelement", ruleVariableCurrentEvent.dataElement)
        assertEquals(org.hisp.dhis.rules.models.RuleValueType.NUMERIC, ruleVariableCurrentEvent.dataElementType)
    }
}
